pkg/controllers/resources/namespaces/syncer.go

```diff
@@ -67,15 +71,19 @@ func (s *namespaceSyncer) Syncer() syncertypes.Sync[client.Object] {
 }
 
 func (s *namespaceSyncer) SyncToHost(ctx *synccontext.SyncContext, event *synccontext.SyncToHostEvent[*corev1.Namespace]) (ctrl.Result, error) {
+	if event.HostOld != nil || event.Virtual.DeletionTimestamp != nil {
```

Comment:
You use HostOld here but haven't enabled ObjectCaching, this seems like a mistake to me

Reply:
yup, thx! added `OptionsProvider` implementation 👍